include/deal.II/base/mpi.h

```diff
@@ -1102,143 +1104,49 @@ namespace Utilities
-    };
+    template <bool...>
+    struct check
+    {};
 
 
 
-    template <>
-    struct is_mpi_type<short>
+    template <typename U, typename... T>
+    constexpr auto contains(std::tuple<T...>)
     {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned char>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned short>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<unsigned long long int>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<float>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<double>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<long double>
-    {
-      static const bool value = true;
-    };
-
-
-
-    template <>
-    struct is_mpi_type<std::complex<float>>
-    {
-      static const bool value = true;
-    };
+      return not std::is_same<
+        std::integer_sequence<bool, false, std::is_same<U, T>::value...>,
+        std::integer_sequence<bool, std::is_same<U, T>::value..., false>>::
+        value;
+    }
 
 
 
-    template <>
-    struct is_mpi_type<std::complex<double>>
+    template <typename T>
+    constexpr bool
+    is_mpi_type()
     {
-      static const bool value = true;
-    };
+      return contains<T>(std::tuple<bool,
+                                    char,
+                                    signed char,
+                                    short,
+                                    int,
+                                    long int,
+                                    unsigned char,
+                                    unsigned short,
+                                    unsigned int,
+                                    unsigned long int,
+                                    unsigned long long int,
+                                    float,
+                                    double,
+                                    long double,
+                                    std::complex<float>,
+                                    std::complex<double>>{});
+    }
 
 
     template <typename T>
-    std::enable_if_t<is_mpi_type<T>::value, T>
+    std::enable_if_t<is_mpi_type<T>(), T>
     broadcast(const MPI_Comm &   comm,
               const T &          object_to_send,
               const unsigned int root_process = 0);
```

Comment:
We should also document this one. Maybe, just by moving it below the other overload and saying something along the lines of "Same as above for built-in MPI types."

Reply:
Fixed in 939abe5